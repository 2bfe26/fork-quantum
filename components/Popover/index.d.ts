import { Component, ReactNode } from 'react';

export interface PopoverProps {
    inverted?: boolean;
    visible?: boolean;
    children: ReactNode[] | ReactNode;
    placement?: 'top' | 'right' | 'bottom' | 'left';
    skin?: 'neutral' | 'primary' | 'success' | 'warning' | 'error';
    trigger: ReactNode;
    onClose?: () => void;
}

export default class Popover extends Component<PopoverProps> {}
