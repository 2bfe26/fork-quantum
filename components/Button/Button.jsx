import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import SocialButtons from './SocialButtons';
import { SIZES } from '../Grid/sub-components/shared/grid-config';
import theme from '../shared/theme';
import skins from './skins';
import Icon from '../Icon/Icon';

const fontSize = ({ size }) => {
  const sizes = {
    xsmall: '12px',
    small: '12px',
    medium: '16px',
    big: '20px',
    xbig: '24px',
  };
  return `font-size: ${sizes[size] || sizes.medium};`;
};

const iconSize = ({ size }) => {
  const sizes = {
    normal: '24px',
  };

  return `font-size: ${sizes[size] || sizes.normal};`;
};

const iconMargin = ({ size }) => {
  const margins = {
    normal: '-5px 5px -7px -5px',
  };

  return `margin: ${margins[size] || margins.normal};`;
};

const padding = ({ size }) => {
  const paddings = {
    xsmall: '4px 12px',
    small: '8px 12px',
    medium: '10px 12px',
    big: '11px 16px',
    xbig: '13px 16px',
  };

  return `padding: ${paddings[size] || paddings.normal};`;
};

const ButtonIcon = styled(Icon)`
  ${iconMargin};
`;

const StyledButton = styled.button`
  border-radius: ${theme.sizes.radius};
  font-weight: bold;
  text-align: center;
  text-decoration: ${props => (props.link ? 'underline' : 'none')};

  ${fontSize}
  ${padding}
  ${theme.mixins.transition()};

  ${ButtonIcon} {
    ${iconSize}
  }

  ${props =>
    props.full &&
    `
    width: 100%;
  `}

  ${props =>
    props.center &&
    `
    display: block;
    margin-left: auto;
    margin-right: auto;
  `}

  ${props => `
    cursor: ${props.disabled ? 'not-allowed' : 'pointer'};
  `}

  ${props => {
    const { unselected, selected, disabled, shadow } = skins(props);

    return `
      background-color: ${
        props.disabled ? disabled.background : unselected.background
      };
      border: 1px solid ${props.disabled ? disabled.border : unselected.border};
      color: ${props.disabled ? disabled.color : unselected.color};
      font-weight: ${
        props.disabled ? disabled.fontWeight : unselected.fontWeight
      };

      ${shadow ? theme.mixins.shadow() : undefined};

      &:active {
        ${shadow && theme.mixins.shadow(2)};
        background-color: ${selected.background};
        border-color: ${selected.border};
        color: ${selected.color};
      }
    `;
  }}

  @media (max-width: ${SIZES.phone}px) {
    margin-left: 0;
    margin-right: 0;
    width: 100%;
  }
`;

class Button extends React.Component {
  static Facebook = SocialButtons.Facebook;

  static Google = SocialButtons.Google;

  render() {
    const { children, icon, size, ...rest } = this.props;

    return (
      <StyledButton {...rest} size={size}>
        {icon && <ButtonIcon size={size} name={icon} />}
        {children}
      </StyledButton>
    );
  }
}

Button.defaultProps = {
  center: false,
  disabled: false,
  full: false,
  icon: '',
  size: 'medium',
  skin: 'primary',
  type: 'button',
  children: 'Catho',
  onClick: () => {},
};

Button.propTypes = {
  center: PropTypes.bool,
  disabled: PropTypes.bool,
  full: PropTypes.bool,
  /** Icon name. The full catalogue can be found
   * [here](/?selectedKind=1.%20Foundation&selectedStory=Icons) */
  icon: PropTypes.string,
  size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'big', 'xbig']),
  skin: PropTypes.oneOf(['primary', 'secondary', 'action']),
  type: PropTypes.oneOf(['button', 'reset', 'submit']),
  children: PropTypes.node,
  onClick: PropTypes.func,
};

export default Button;
