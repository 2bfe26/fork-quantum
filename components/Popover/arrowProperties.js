import Colors from '../Colors';

const ARROW_SIZE = '16px';
const SIDE_POSITION_X = -8;
const CENTER_POSITION = '50%';

const getColorBySkin = (
  skin,
  {
    components: {
      popover: { skins },
    },
  },
  inverted,
) => {
  if (inverted) {
    return skins[skin].text;
  }
  return skins[skin].background;
};

const placementPosition = {
  top: `
    content: '▼';
    left: ${CENTER_POSITION};
    text-shadow: 0px 4px 4px ${Colors.SHADOW[50]};
    bottom: 0;
    transform: translateX(-${CENTER_POSITION});
  `,
  bottom: `
    top: -4px;
    content: '▲';
    left: ${CENTER_POSITION};
    text-shadow: 0px -3px 4px ${Colors.SHADOW[50]};
    transform: translateX(-${CENTER_POSITION});
  `,
  right: `
    bottom: ${CENTER_POSITION};
    content: '◀';
    left: ${SIDE_POSITION_X}px;
    text-shadow: -3px 1px 4px ${Colors.SHADOW[50]};
  `,
  left: `
    bottom: ${CENTER_POSITION};
    content: '▶';
    right: ${SIDE_POSITION_X}px;
    text-shadow: 3px 1px 4px ${Colors.SHADOW[50]};
  `,
};

const getArrow = (placement, skin, theme, inverted) => `
  color: ${getColorBySkin(skin, theme, inverted)}
  font-size: ${ARROW_SIZE};
  position: absolute;

  ${placementPosition[placement]}
`;

export default getArrow;
