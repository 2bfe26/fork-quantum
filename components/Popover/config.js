import Colors from '../Colors';

const ARROW_SIZE = '16px';
const ARROW_POSITION_Y = '101%';
const SIDE_POSITION_X = -10;
const CENTER_POSITION = '50%';

const getColorBySkin = skin => {
  const indexColor = skin.toUpperCase();
  return skin === 'default' ? Colors.WHITE : Colors[indexColor][200];
};

const getArrow = (placement, skin) => {
  let placementPosition = '';
  const placementPositionTop = `
  content: '▼';
  left: ${CENTER_POSITION};
  text-shadow: 0px 4px 4px ${Colors.SHADOW[50]};
  top: ${ARROW_POSITION_Y};
  transform: translateX(-${CENTER_POSITION});`;

  switch (placement) {
    case 'top':
      placementPosition = placementPositionTop;
      break;
    case 'bottom':
      placementPosition = `
      bottom: ${ARROW_POSITION_Y};
      content: '▲';
      left: ${CENTER_POSITION};
      text-shadow: 0px -3px 4px ${Colors.SHADOW[50]};
      transform: translateX(-${CENTER_POSITION});`;
      break;
    case 'right':
      placementPosition = `
      bottom: ${CENTER_POSITION};
      content: '◀';
      left: ${SIDE_POSITION_X}px;
      text-shadow: -3px 1px 4px ${Colors.SHADOW[50]};`;
      break;
    case 'left':
      placementPosition = `
      bottom: ${CENTER_POSITION};
      content: '▶';
      right: ${SIDE_POSITION_X}px;
      text-shadow: 3px 1px 4px ${Colors.SHADOW[50]};`;
      break;
    default:
      placementPosition = placementPositionTop;
  }

  return `
  color: ${getColorBySkin(skin)}
  font-size: ${ARROW_SIZE};
  position: absolute;
  
  ${placementPosition}
  `;
};

export default getArrow;
