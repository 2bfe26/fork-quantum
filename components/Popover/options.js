import Colors from '../Colors';

const ARROW_SIZE = '8px';
const CORNER_ARROW_DISTANCE = 25;

const skins = {
  p2p: Colors.P2P['500'],
};

const upDownBorders = `
  border-left: ${ARROW_SIZE} solid transparent;
  border-right: ${ARROW_SIZE} solid transparent;
  left: 50%;
  transform: translateX(-50%);
`;

const sideBorders = `
  border-top: ${ARROW_SIZE} solid transparent;
  border-bottom: ${ARROW_SIZE} solid transparent;
  top: 50%;
  transform: translateY(-50%);
`;

const arrowPosition = (position) => {
  const positions = {
    bottom: `
      ${upDownBorders}
      border-bottom: ${ARROW_SIZE} solid;
      border-bottom-color: inherit;
      top: -${ARROW_SIZE};
    `,
    top: `
      ${upDownBorders}
      border-top: ${ARROW_SIZE} solid;
      border-top-color: inherit;
      bottom: -${ARROW_SIZE};
    `,
    left: `
      ${sideBorders}
      border-left: ${ARROW_SIZE} solid;
      border-left-color: inherit;
      right: -${ARROW_SIZE};
    `,
    right: `
      ${sideBorders}
      border-right: ${ARROW_SIZE} solid;
      border-right-color: inherit;
      left: -${ARROW_SIZE};
    `,
    'top-left': `
      ${upDownBorders}
      border-top: ${ARROW_SIZE} solid;
      border-top-color: inherit;
      bottom: -${ARROW_SIZE};

      right: ${CORNER_ARROW_DISTANCE}px;
      transform: translateX(50%);
      left: auto;
    `,
    'top-right': `
      ${upDownBorders}
      border-top: ${ARROW_SIZE} solid;
      border-top-color: inherit;
      bottom: -${ARROW_SIZE};

      left: ${CORNER_ARROW_DISTANCE}px;
      transform: translateX(-50%);
    `,
  };

  return positions[position];
};

const popoverPosition = ({
  place, popoverHeight, popoverWidth, childrenWidth,
}) => {
  const position = {
    top: `top: -${popoverHeight + 15}px; left: 50%; margin-left: -${Math.floor(popoverWidth / 2)}px;`,
    right: `right: -${popoverWidth + 15}px; top: 50%; margin-top: -${Math.floor(popoverHeight / 2)}px;`,
    left: `left: -${popoverWidth + 15}px; top: 50%; margin-top: -${Math.floor(popoverHeight / 2)}px;`,
    'top-left': `top: -${popoverHeight + 15}px; left: ${Math.floor(childrenWidth / 2) + CORNER_ARROW_DISTANCE}px; margin-left: -${popoverWidth}px;`,
    'top-right': `top: -${popoverHeight + 15}px; left: -${Math.floor(childrenWidth / 2) + CORNER_ARROW_DISTANCE}px; margin-left: 100%;`,
  };

  return position[place] || position.top;
};

export default {
  skins, arrowPosition, popoverPosition,
};
