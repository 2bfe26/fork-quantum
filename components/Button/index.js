import { Button, ButtonSimple } from './Button';

export { Button, ButtonSimple };
