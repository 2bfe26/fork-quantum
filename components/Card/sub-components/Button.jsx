import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {default as OriginalButton} from '../../Button';

const ButtonWrapper = styled(OriginalButton)`
  background-color: transparent;
  padding: 0;
  margin: 0 10px 0 0;
  box-shadow: none;
  font-family: oxygen;
  text-transform: uppercase;
  font-size: 12px;

  &:active {
    box-shadow: none;
    background-color: transparent;
    border: 1px solid transparent;
  }
`;

const Button = props => <ButtonWrapper {...props} />;

export default Button;