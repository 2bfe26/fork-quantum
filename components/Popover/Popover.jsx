import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Colors from '../Colors';
import Icon from '../Icon';
import Button from '../Button';

const padding = '18px 24px';

const arrowSize = '8px';
const cornerArrowDistance = 25;
const upDownBorders = `
  border-left: ${arrowSize} solid transparent;
  border-right: ${arrowSize} solid transparent;
  left: 50%;
  transform: translateX(-50%);
`;

const sideBorders = `
  border-top: ${arrowSize} solid transparent;
  border-bottom: ${arrowSize} solid transparent;
  top: 50%;
  transform: translateY(-50%);
`;

const placement = {
  arrowPosition: {
    bottom: `
      ${upDownBorders}
      border-bottom: ${arrowSize} solid;
      border-bottom-color: inherit;
      top: -${arrowSize};
    `,
    top: `
      ${upDownBorders}
      border-top: ${arrowSize} solid;
      border-top-color: inherit;
      bottom: -${arrowSize};
    `,
    left: `
      ${sideBorders}
      border-left: ${arrowSize} solid;
      border-left-color: inherit;
      right: -${arrowSize};
    `,
    right: `
      ${sideBorders}
      border-right: ${arrowSize} solid;
      border-right-color: inherit;
      left: -${arrowSize};
    `,
    'top-left': `
      ${upDownBorders}
      border-top: ${arrowSize} solid;
      border-top-color: inherit;
      bottom: -${arrowSize};

      right: ${cornerArrowDistance}px;
      transform: translateX(50%);
      left: auto;
    `,
    'top-right': `
      ${upDownBorders}
      border-top: ${arrowSize} solid;
      border-top-color: inherit;
      bottom: -${arrowSize};

      left: ${cornerArrowDistance}px;
      transform: translateX(-50%);
    `,
  },

  popoverPosition: ({
    place, popoverHeight, popoverWidth, childrenWidth,
  }) => {
    const position = {
      top: `top: -${popoverHeight + 20}px; left: 50%; margin-left: -${Math.floor(popoverWidth / 2)}px;`,
      right: `right: -${popoverWidth + 25}px; top: 50%; margin-top: -${Math.floor(popoverHeight / 2)}px;`,
      left: `left: -${popoverWidth + 25}px; top: 50%; margin-top: -${Math.floor(popoverHeight / 2)}px;`,
      'top-left': `top: -${popoverHeight + 20}px; left: ${Math.floor(childrenWidth / 2) + cornerArrowDistance}px; margin-left: -${popoverWidth}px;`,
      'top-right': `top: -${popoverHeight + 20}px; left: -${Math.floor(childrenWidth / 2) + cornerArrowDistance}px; margin-left: 100%;`,
    };

    return position[place] || position.top;
  },
};

const Wrapper = styled.div`
  position: relative;
`;

const PopoverContainer = styled.div`
  background-color: ${Colors.WHITE};
  border-color: ${Colors.WHITE};
  border-radius: 8px;
  box-shadow: 0 3px 3px 0 rgba(0, 0, 0, 0.2),
              0 8px 14px 3px rgba(0, 0, 0, 0.12),
              0 8px 10px 1px rgba(0, 0, 0, 0.14);

  cursor: default;
  position: absolute;
  z-index: 100;
  opacity: ${({ show }) => (show ? '1' : '0')};
  transition: opacity 0.2s ease-in-out;

  ${placement.popoverPosition}

  &:before {
    content: '';
    position: absolute;
    ${({ place }) => placement.arrowPosition[place]};
  }
`;

const Title = styled.div`
  align-items: center;
  display: flex;
  justify-content: ${({ title }) => (title ? 'space-between' : 'flex-end')};
  padding: ${padding};

  ${({ title }) => (title
    ? `border-bottom: 1px solid ${Colors.GREY['50']};`
    : 'padding-bottom: 0;'
  )};
`;

const Content = styled.div`
  padding: ${padding};
`;

const CloseIcon = styled(Icon)`
  cursor: pointer;
`;

const ChildrenContainer = styled.div`
  cursor: pointer;
`;

class Popover extends Component {
  constructor(props) {
    super(props);

    this.state = {
      show: false,
      popoverMeasures: {
        width: null,
        height: null,
      },
      childrenMeasures: {
        width: null,
        height: null,
      },
    };
  }

  componentDidMount() {
    this.measure();
  }

  shouldComponentUpdate(nextProps, nextState) {
    return (
      this.props.title !== nextProps.title ||
      this.props.content !== nextProps.content ||
      this.props.place !== nextProps.place ||
      this.state.show !== nextState.show ||
      this.state.popoverMeasures.width !== nextState.popoverMeasures.width ||
      this.state.popoverMeasures.height !== nextState.popoverMeasures.height ||
      this.state.childrenMeasures.width !== nextState.childrenMeasures.width ||
      this.state.childrenMeasures.height !== nextState.childrenMeasures.height
    );
  }

  componentDidUpdate() {
    this.measure();
  }

  measure() {
    const { clientWidth: popoverWidth, clientHeight: popoverHeight } = this.popoverRef;
    const { clientWidth: childrenWidth, clientHeight: childrenHeight } = this.childrenRef;

    this.setState({
      popoverMeasures: {
        width: popoverWidth,
        height: popoverHeight,
      },
      childrenMeasures: {
        width: childrenWidth,
        height: childrenHeight,
      },
    });
  }

  toggleVisibility = (event) => {
    event.preventDefault();

    const { show } = this.state;

    this.setState({ show: !show });
  }

  hide = () => {
    this.setState({ show: false });
  }

  render() {
    const {
      title, content, children, place, closeTitle,
    } = this.props;

    const {
      show,
      popoverMeasures: {
        width: popoverWidth,
        height: popoverHeight,
      },
      childrenMeasures: {
        width: childrenWidth,
        height: childrenHeight,
      },
    } = this.state;

    return (
      <Wrapper>
        <PopoverContainer
          place={place}
          popoverWidth={popoverWidth}
          popoverHeight={popoverHeight}
          childrenWidth={childrenWidth}
          childrenHeight={childrenHeight}
          show={show}
          innerRef={(domElement) => { this.popoverRef = domElement; }}
        >
          <Title title={title}>
            { title && <span>{ title }</span> }
            <CloseIcon name="close" onClick={this.hide} title={closeTitle} />
          </Title>

          <Content>{ content }</Content>
        </PopoverContainer>

        <ChildrenContainer
          onClick={this.toggleVisibility}
          innerRef={(domElement) => { this.childrenRef = domElement; }}
        >
          { children }
        </ChildrenContainer>
      </Wrapper>
    );
  }
}

Popover.propTypes = {
  title: PropTypes.string,
  content: PropTypes.node,
  children: PropTypes.node,
  closeTitle: PropTypes.string,
  place: PropTypes.oneOf([
    'top',
    'top-right',
    'top-left',
    'right',
    'left',
  ]),
};

Popover.defaultProps = {
  title: 'Title',
  content: (
    <React.Fragment>
      <div style={{ width: '300px' }}>Content</div>
      <div style={{ width: '300px' }}>Content</div>
      <div style={{ width: '300px' }}>Content</div>
      <div style={{ width: '300px' }}>Content</div>
      <div style={{ width: '300px' }}>Content</div>
      <div style={{ width: '300px' }}>Content</div>
    </React.Fragment>
  ),
  children: <Button size="big">123456789</Button>,
  place: 'top',
  closeTitle: 'Fechar',
};

export default Popover;
