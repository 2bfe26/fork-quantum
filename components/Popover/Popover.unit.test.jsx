import React from 'react';
import { mount } from 'enzyme';
import Popover from './Popover';
import Button from './../Button';
import Icon from './../Icon';

describe('Popover', () => {
  it('should match the snapshot', () => {
    const examples = [
      (
        <Popover>Interact</Popover>
      ), (
        <Popover title="Title" closeTitle="Close">
          Interact
        </Popover>
      ), (
        <Popover
          content={(
            <div>
              <h1>My title</h1>
              <ul>
                <li>Content</li>
                <li>Content</li>
                <li>Content</li>
              </ul>

              <hr />

              <Button>Click me</Button>
            </div>
          )}
        >
          Interact
        </Popover>
      ), (
        <Popover place="top-right">
          Interact
        </Popover>
      ), (
        <Popover place="top-left">
          Interact
        </Popover>
      ), (
        <Popover place="left">
          Interact
        </Popover>
      ), (
        <Popover skin="p2p">
          Interact
        </Popover>
      ), (
        <Popover ribbon="Limited edition">
          Interact
        </Popover>
      ),
    ];

    examples.forEach((component) => {
      expect(mount(component).html()).toMatchSnapshot();
    });
  });

  it('should appear when children is clicked', () => {
    const wrapper = mount((
      <Popover>
        <Button>Interact</Button>
      </Popover>
    ));

    expect(wrapper.state('show')).toBe(false);

    wrapper.find(Button).simulate('click', {});

    expect(wrapper.state('show')).toBe(true);
  });

  describe('should disappear according matched trigger', () => {
    const wrapper = mount((
      <Popover>
        <Button>Interact</Button>
      </Popover>
    ));

    beforeEach(() => {
      wrapper.find(Button).simulate('click', {});
    });

    it('when children is clicked twice', () => {
      wrapper.find(Button).simulate('click', {});
      expect(wrapper.state('show')).toBe(false);
    });

    it('when close icon is clicked', () => {
      wrapper.find(Icon).simulate('click', {});
      expect(wrapper.state('show')).toBe(false);
    });

    it('when a click outside popover is made', () => {
      const eventMap = {};

      // Add a clickable node to document
      const buttonNode = document.createElement('button');
      buttonNode.id = 'new-button';
      buttonNode.innerText = 'New Button';
      document.body.appendChild(buttonNode);

      // Populate eventMap
      document.addEventListener = jest.fn((event, cb) => {
        eventMap[event] = cb;
      });

      const popoverWrapper = mount(<Popover show>Interact</Popover>);

      // Trigger outside click handler
      eventMap.mousedown({ target: buttonNode });

      expect(popoverWrapper.state('show')).toBe(false);
    });
  });
});
