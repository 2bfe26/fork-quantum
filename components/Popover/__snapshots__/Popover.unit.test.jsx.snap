// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover should match the snapshot 1`] = `"<div class=\\"Popover__Wrapper-kBCVMN eTEEdZ\\"><div class=\\"Popover__PopoverContainer-cmbqyT hdLPme\\"><div class=\\"Popover__Header-iBemXC ceFmJK\\" title=\\"\\"><span class=\\"material-icons MuiIcon-root-1 Popover__CloseIcon-gRRMUz jAemtJ\\" aria-hidden=\\"true\\" title=\\"\\">close</span></div><div class=\\"Popover__Content-kzlOSc cUiUGK\\"></div></div><div class=\\"Popover__ChildrenContainer-fJnAhL csqLPs\\">Interact</div></div>"`;

exports[`Popover should match the snapshot 2`] = `"<div class=\\"Popover__Wrapper-kBCVMN eTEEdZ\\"><div class=\\"Popover__PopoverContainer-cmbqyT hdLPme\\"><div class=\\"Popover__Header-iBemXC kSIykf\\" title=\\"Title\\"><span>Title</span><span class=\\"material-icons MuiIcon-root-1 Popover__CloseIcon-gRRMUz jAemtJ\\" aria-hidden=\\"true\\" title=\\"Close\\">close</span></div><div class=\\"Popover__Content-kzlOSc cUiUGK\\"></div></div><div class=\\"Popover__ChildrenContainer-fJnAhL csqLPs\\">Interact</div></div>"`;

exports[`Popover should match the snapshot 3`] = `"<div class=\\"Popover__Wrapper-kBCVMN eTEEdZ\\"><div class=\\"Popover__PopoverContainer-cmbqyT hdLPme\\"><div class=\\"Popover__Header-iBemXC ceFmJK\\" title=\\"\\"><span class=\\"material-icons MuiIcon-root-1 Popover__CloseIcon-gRRMUz jAemtJ\\" aria-hidden=\\"true\\" title=\\"\\">close</span></div><div class=\\"Popover__Content-kzlOSc cUiUGK\\"><div><h1>My title</h1><ul><li>Content</li><li>Content</li><li>Content</li></ul><hr><button class=\\"Button__StyledButton-jJDkJK YtayR\\" type=\\"button\\">Click me</button></div></div></div><div class=\\"Popover__ChildrenContainer-fJnAhL csqLPs\\">Interact</div></div>"`;

exports[`Popover should match the snapshot 4`] = `"<div class=\\"Popover__Wrapper-kBCVMN eTEEdZ\\"><div class=\\"Popover__PopoverContainer-cmbqyT ftNyoz\\"><div class=\\"Popover__Header-iBemXC ceFmJK\\" title=\\"\\"><span class=\\"material-icons MuiIcon-root-1 Popover__CloseIcon-gRRMUz jAemtJ\\" aria-hidden=\\"true\\" title=\\"\\">close</span></div><div class=\\"Popover__Content-kzlOSc cUiUGK\\"></div></div><div class=\\"Popover__ChildrenContainer-fJnAhL csqLPs\\">Interact</div></div>"`;

exports[`Popover should match the snapshot 5`] = `"<div class=\\"Popover__Wrapper-kBCVMN eTEEdZ\\"><div class=\\"Popover__PopoverContainer-cmbqyT dGEDYF\\"><div class=\\"Popover__Header-iBemXC ceFmJK\\" title=\\"\\"><span class=\\"material-icons MuiIcon-root-1 Popover__CloseIcon-gRRMUz jAemtJ\\" aria-hidden=\\"true\\" title=\\"\\">close</span></div><div class=\\"Popover__Content-kzlOSc cUiUGK\\"></div></div><div class=\\"Popover__ChildrenContainer-fJnAhL csqLPs\\">Interact</div></div>"`;

exports[`Popover should match the snapshot 6`] = `"<div class=\\"Popover__Wrapper-kBCVMN eTEEdZ\\"><div class=\\"Popover__PopoverContainer-cmbqyT kahOYe\\"><div class=\\"Popover__Header-iBemXC ceFmJK\\" title=\\"\\"><span class=\\"material-icons MuiIcon-root-1 Popover__CloseIcon-gRRMUz jAemtJ\\" aria-hidden=\\"true\\" title=\\"\\">close</span></div><div class=\\"Popover__Content-kzlOSc cUiUGK\\"></div></div><div class=\\"Popover__ChildrenContainer-fJnAhL csqLPs\\">Interact</div></div>"`;

exports[`Popover should match the snapshot 7`] = `"<div class=\\"Popover__Wrapper-kBCVMN eTEEdZ\\"><div class=\\"Popover__PopoverContainer-cmbqyT fiNskW\\"><div class=\\"Popover__Header-iBemXC ceFmJK\\" title=\\"\\"><span class=\\"material-icons MuiIcon-root-1 Popover__CloseIcon-gRRMUz jAemtJ\\" aria-hidden=\\"true\\" title=\\"\\">close</span></div><div class=\\"Popover__Content-kzlOSc cUiUGK\\"></div></div><div class=\\"Popover__ChildrenContainer-fJnAhL csqLPs\\">Interact</div></div>"`;

exports[`Popover should match the snapshot 8`] = `"<div class=\\"Popover__Wrapper-kBCVMN eTEEdZ\\"><div class=\\"Popover__PopoverContainer-cmbqyT hdLPme\\"><div class=\\"Ribbon__Wrapper-eQgmUA coIDOu\\"></div><div class=\\"Popover__Header-iBemXC ceFmJK\\" title=\\"\\"><span class=\\"material-icons MuiIcon-root-1 Popover__CloseIcon-gRRMUz jAemtJ\\" aria-hidden=\\"true\\" title=\\"\\">close</span></div><div class=\\"Popover__Content-kzlOSc jjfYuj\\"></div></div><div class=\\"Popover__ChildrenContainer-fJnAhL csqLPs\\">Interact</div></div>"`;
